adl/stdlib/sys/types.adl-hs

```diff
@@ -39,6 +39,18 @@ annotation Error HaskellCustomType {
     ]
   };
 
+annotation Result HaskellCustomType {
+  "haskellname" : "Result",
+  "haskellimports" : [],
+  "insertCode" : ["type Result a b = Prelude.Either a b"],
```

Comment:
I think this is incorrect. From:

http://hackage.haskell.org/package/base-4.14.0.0/docs/Data-Either.html

> by convention, the Left constructor is used to hold an error value and the Right constructor is used to hold a correct value (mnemonic: "right" also means "correct").

This convention is baked into the monad implementation for Either.